CONTRIBUTING.md

```diff
@@ -1,60 +1,129 @@
+* Fork and clone the repo.
+* Setup your preferred IDE (IntelliJ, VSCode, etc.)
 
-## Acceptance tests
+Requirements:
+* [Terraform](https://www.terraform.io/downloads.html) >= 1.0.0
+* [Go](https://golang.org/doc/install) >= 1.25
+* Docker (for acceptance tests)
 
-```bash
-make docker-testacc
-```
-
-Run a single test with terraform debug enabled:
-```bash
-env TF_LOG=DEBUG make docker-testacc TESTARGS='-run ^TestAccResourceDataStreamLifecycle$$'
-```
-
-A way to forward debug logs to a file:
-```bash
-env TF_ACC_LOG_PATH=/tmp/tf.log TF_ACC_LOG=DEBUG TF_LOG=DEBUG make docker-testacc
-```
+## Development Workflow
 
+* Create a new branch for your changes.
+* Make your changes. See [Useful Commands](#useful-commands) and [Debugging](#running--debugging-the-provider).
+* Validate your changes
+  * Run unit and acceptance tests (See [Running Acceptance Tests](#running-acceptance-tests)).
+  * Run `make lint` to check linting and formatting. For this check to succeed, all changes must have been committed.
+  * All checks also run automatically on every PR.
+* Submit your PR for review.
+* Add a changelog entry in `CHANGELOG.md` under the `Unreleased` section. This will be included in the release notes of the next release. The changelog entry references the PR, so it has to be added after the PR has been opened.
 
-## Update documentation
+When creating new resources:
+* Use the [Plugin Framework](https://developer.hashicorp.com/terraform/plugin/framework/getting-started/code-walkthrough) for new resources.
+  * Use an existing resource (e.g. `internal/elasticsearch/security/system_user`) as a template.
+  * Some resources use the deprecated Terraform SDK, so only resources using the new Terraform Framework should be used as reference.
+* Use the generated API clients to interact with the Kibana APIs. (See [Working with Generated API Clients](#working-with-generated-api-clients)
+* Add a documentation template and examples for the resource. See [Updating Documentation](#updating-documentation) for more details.
+* Write unit and acceptance tests.
 
-Update documentation templates in `./templates` directory and re-generate docs via:
-```bash
-make docs-generate
-```
+### Useful Commands
 
-## Update `./CHANGELOG.md`
+* `make build`: Build the provider.
+* `make lint`: Lints and formats the code.
+* `make test`: Run unit tests.
+* `make docs-generate`: Generate documentation.
 
-List of previous commits is a good example of what should be included in the changelog.
+### Running & Debugging the Provider
 
+Run the provider in debug mode and reattach the provider in Terraform:
+* Launch `main.go` with the `-debug` flag from your IDE.
+* After launching, the provider will print an env var. Copy the printed `TF_REATTACH_PROVIDERS='{…}'` value.
+* Export it in your shell where you run Terraform: `export TF_REATTACH_PROVIDERS='{…}'`.
+* Terraform will now talk to your debug instance, and you can set breakpoints.
```

Comment:
Maybe it makes sense to refer also to some relevant TF docs - https://developer.hashicorp.com/terraform/plugin/debugging#starting-a-provider-in-debug-mode.